c_src/sqlite3_nif.c

```diff
@@ -594,7 +585,7 @@ make_cell(ErlNifEnv* env, sqlite3_stmt* statement, unsigned int i)
               sqlite3_column_bytes(statement, i));
 
         default:
-            return make_atom(env, "unsupported");
+            return am_nil;
```

Comment:
This technically doesn't matter since there are only five types in SQLite.

Reply:
I don't think it is possible to reach this state, I defined it _just in case_ it did happen. I don't know if extensions can even set this to a custom type.